### DISCUSSION TITLE
Nullable DateOnly fields - is there a reason they're expanded?

### DISCUSSION THREAD

**Discussion by ItsRobbAllen** [ravendb/ravendb#13944] [General]
If I have a class with 
```
public class MyClass { ...
public DateOnly Expiration {get;set;}
...}
```
It gets stored as 
```{...
"Expiration": "2018-02-25"
...{
```

But if the property is nullable
```public DateOnly? Expiration {get;set;}```

Then it looks like this
```
"ExpirationDate": {
                "Year": 2018,
                "Month": 2,
                "Day": 25,
                "DayOfWeek": "Sunday",
                "DayOfYear": 56,
                "DayNumber": 736749
            }
```

Is there a reason for this that I'm missing?

**Comment by haludi** [comment]
Hi

This is all about the `DateOnly.ToString`  - https://docs.microsoft.com/en-us/dotnet/api/system.dateonly.tostring?view=net-6.0#:~:text=specific%20format%20information.-,ToString(),be%20formatted%20in%20short%20form.
https://stackoverflow.com/questions/3025361/c-sharp-datetime-to-yyyymmddhhmmss-format

We are using `Newtonsoft.Json` and you can create your own converter:
```
public class MyClass 
{ 
    public DateOnly Expiration {get;set;}
}


private class DateOnlyConverter : JsonConverter
{
    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
    {
        var m = ((DateOnly)value);
        writer.WriteValue(m.ToString("M MM MMM MMMM"));
    }

    public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
    {
        //Your logic
        return new DateOnly
        {
        };
    }

    public override bool CanConvert(Type objectType)
    {
        return objectType == typeof(DateOnly);
    }
}


[Fact]
public async Task Test()
{
    using var store = GetDocumentStore(new Options
    {
        ModifyDocumentStore = s =>
        {
            s.Conventions = new DocumentConventions
            {
                Serialization = new NewtonsoftJsonSerializationConventions {CustomizeJsonSerializer = s => s.Converters.Add(new DateOnlyConverter())}
            };
        }
    });
    using (var session = store.OpenAsyncSession())
    {
        await session.StoreAsync(new MyClass{Expiration = new DateOnly(1985, 6, 17)});
        await session.SaveChangesAsync();
    }
}
```

Best regards,
Igal

**Reply by mcrio** [reply]
Hi @haludi so it's expected that by default a nullable `DateOnly` will be stored as some unexpected structure and not `null`? (although from the example I don't understand what the `ExpirationDate` property is belonging to)

**Reply by ppekrol** [reply]
Hi @mcrio 
This is a bug. We will be fixing that. Thanks for your feedback.

**Reply by ppekrol** [reply]
I've created an issue for this: https://issues.hibernatingrhinos.com/issue/RavenDB-18399

**Comment by ayende** [comment]
This is fixed here, I believe: https://github.com/ravendb/ravendb/pull/13947#pullrequestreview-925880647
